plugins/inputs/odyssey/odyssey.go

```diff
@@ -0,0 +1,165 @@
+package odyssey
+
+import (
+	"bytes"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/postgresql"
+	_ "github.com/jackc/pgx/stdlib" // register driver
+)
+
+type Odyssey struct {
+	postgresql.Service
+	IncludeQuery []string `toml:"include_query"`
+}
+
+var ignoredColumns = map[string]bool{"user": true, "database": true, "pool_mode": true,
+	"avg_req": true, "avg_recv": true, "avg_sent": true, "avg_query": true,
+}
+
+var sampleConfig = `
+  ## specify address via a url matching:
+  ##   postgres://[console[:password]]@localhost[/dbname]\
+  ##       ?sslmode=[disable|verify-ca|verify-full]
+  ## or a simple string:
+  ##   host=localhost user=console password=... sslmode=... dbname=app_production
+  ##
+  ## All connection parameters are optional.
+  ##
+  address = "host=localhost user=postgresql sslmode=disable dbname=console port=6432"
+
+  ## Default queries - ['SHOW_POOLS', 'SHOW_ERRORS', 'SHOW_CLIENTS', 'SHOW_STATS']
+  ## A list of queries to include. If not specified, all the above listed queries are used.
+  include_query = []
+`
+
+func (p *Odyssey) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Odyssey) Description() string {
+	return "Read metrics from Odyssey"
+}
+
+func (p *Odyssey) Gather(acc telegraf.Accumulator) error {
+	var (
+		err     error
+		query   string
+		queries []string
+		columns []string
+	)
+
+	queries = []string{`SHOW POOLS`, `SHOW CLIENTS`, `SHOW ERRORS`, `SHOW STATS`}
+
+	if len(p.IncludeQuery) > 0 {
+		queries = p.IncludeQuery
+	}
+
+	for _, value := range queries {
+		query = value
+
+		rows, err := p.DB.Query(query)
+		if err != nil {
+			return err
+		}
+
+		defer rows.Close()
+
+		// grab the column information from the result
+		if columns, err = rows.Columns(); err != nil {
+			return err
+		}
+
+		for rows.Next() {
+			tags, columnMap, err := p.accRow(rows, acc, columns)
+
+			if err != nil {
+				return err
+			}
+
+			if user, ok := columnMap["user"]; ok {
+				if s, ok := (*user).(string); ok && s != "" {
+					tags["user"] = s
+				}
+			}
+
+			if poolMode, ok := columnMap["pool_mode"]; ok {
+				if s, ok := (*poolMode).(string); ok && s != "" {
+					tags["pool_mode"] = s
+				}
+			}
+
+			fields := make(map[string]interface{})
+			for col, val := range columnMap {
+				_, ignore := ignoredColumns[col]
+				if !ignore {
+					fields[col] = *val
+				}
+			}
+			acc.AddFields(query, fields, tags)
+		}
+	}
+
+	return err
+}
+
+type scanner interface {
+	Scan(dest ...interface{}) error
+}
+
+func (p *Odyssey) accRow(row scanner, acc telegraf.Accumulator, columns []string) (map[string]string,
+	map[string]*interface{}, error) {
+	var columnVars []interface{}
+	var dbname bytes.Buffer
```

Comment:
you're using a bytes.Buffer to assign convert strings into and then converting it back to a string at the end. I'd suggest just using a string lol